raqm.c

```diff
@@ -61,320 +62,440 @@ static const FriBidiChar paired_chars[] = {
-raqm_glyph_info_t *raqm_shape(const char *text , FT_Face face, raqm_direction_t direction) {
+raqm_glyph_info_t*
+raqm_shape (const char* text,
+            FT_Face face,
+            raqm_direction_t direction)
+{
     int i = 0;
-    const char *str = text;
-    FriBidiStrIndex size = strlen(str);
+    const char* str;
+    FriBidiStrIndex size;
+    FriBidiChar* unicode_str;
+    FriBidiCharType* types;
+    FriBidiLevel* levels;
+    FriBidiParType par_type;
+    hb_script_t* scripts;
+    hb_unicode_funcs_t* unicode_funcs;
+    hb_script_t lastScriptValue;
+    int lastScriptIndex = -1;
+    int lastSetIndex = -1;
+    Stack* script_stack;
+    int run_count;
+    Run* run;
+    hb_font_t* hb_font;
+    unsigned int glyph_count;
+    int total_glyph_count = 0;
+    unsigned int postion_length;
+    hb_glyph_info_t* hb_glyph_info;
+    hb_glyph_position_t* hb_glyph_position;
+    raqm_glyph_info_t* glyph_info;
+    str = text;
+    size = strlen (str);
 
-    FriBidiChar *uni_str = (FriBidiChar*) malloc (sizeof (FriBidiChar) * size);
-    FriBidiStrIndex length = fribidi_charset_to_unicode (FRIBIDI_CHAR_SET_UTF8, str, size, uni_str);
+    unicode_str = (FriBidiChar*) malloc (sizeof (FriBidiChar) * size);
+    FriBidiStrIndex length = fribidi_charset_to_unicode (FRIBIDI_CHAR_SET_UTF8, str, size, unicode_str);
 
-    FriBidiCharType *types = (FriBidiCharType*) malloc (sizeof (FriBidiCharType) * length);
-    FriBidiLevel *levels = (FriBidiLevel*) malloc (sizeof (FriBidiLevel) * length);
-    fribidi_get_bidi_types (uni_str, length, types);
+    types = (FriBidiCharType*) malloc (sizeof (FriBidiCharType) * length);
+    levels = (FriBidiLevel*) malloc (sizeof (FriBidiLevel) * length);
+    fribidi_get_bidi_types (unicode_str, length, types);
 
-    FriBidiParType par_type = FRIBIDI_PAR_ON;
+    par_type = FRIBIDI_PAR_ON;
     if (direction == RAQM_DIRECTION_RTL)
+    {
         par_type = FRIBIDI_PAR_RTL;
+    }
     else if (direction == RAQM_DIRECTION_LTR)
+    {
         par_type = FRIBIDI_PAR_LTR;
+    }
 
-    TEST("Text is: %s\n\n", text);
+    TEST ("Text is: %s\n\n", text);
 
     fribidi_get_par_embedding_levels (types, length, &par_type, levels);
 
     /* Handeling script detection for each character of the input string,
-	if the character script is common or inherited it takes the script
-	of the character before it except some special paired characters */
-    hb_script_t *scripts = (hb_script_t*) malloc (sizeof (hb_script_t) * length);
-    hb_unicode_funcs_t *ufuncs = hb_unicode_funcs_get_default();
-
-    TEST("Before script detection:\n");
-    for (i = 0; i < length; ++i) {
-	scripts[i] = hb_unicode_script(ufuncs, uni_str[i]);
-#ifdef TESTING
-        SCRIPT_TO_STRING(scripts[i]);
-	TEST("script for ch[%d]\t%s\n",i ,buff);
-#endif
+    if the character script is common or inherited it takes the script
+    of the character before it except some special paired characters */
```

Comment:
Please align the text inside the comment to the first character (like the old code).